lib/modem_info/modem_info.c

```diff
@@ -720,6 +720,119 @@ int modem_info_rsrp_register(rsrp_cb_t cb)
 	return 0;
 }
 
+/* FW UUID is 36 characters: XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX */
+#define FW_UUID_SIZE 37
+
+int modem_info_get_fw_uuid(char *buf, size_t buf_size)
+{
+	int ret;
+
+	if (buf_size < FW_UUID_SIZE) {
+		return -EINVAL;
+	}
+
+	ret = nrf_modem_at_scanf("AT%XMODEMUUID",
+				 "%%XMODEMUUID: %" STRINGIFY(FW_UUID_SIZE) "[^\r\n]",
+				 buf);
+	if (ret != 1) {
+		LOG_ERR("Could not get FW ID, error: %d", ret);
+		return -EIO;
+	}
+	return 0;
+}
+
+/* The "short software identification" has no specified size limit. */
+#define FW_VERSION_SIZE 32
+
+int modem_info_get_fw_version(char *buf, size_t buf_size)
+{
+	int ret;
+
+	if (buf_size < FW_VERSION_SIZE) {
+		return -EINVAL;
+	}
+
+	ret = nrf_modem_at_scanf("AT%SHORTSWVER",
+				 "%%SHORTSWVER: %" STRINGIFY(FW_VERSION_SIZE) "[^\r\n]",
+				 buf);
+
+	if (ret != 1) {
+		LOG_ERR("Could not get FW version, error: %d", ret);
+		return -EIO;
+	}
+	return 0;
+}
+
+/* SVN is always 2 digits: XX */
+#define SVN_SIZE 3
+
+int modem_info_get_svn(char *buf, size_t buf_size)
+{
+	int ret;
+
+	if (buf_size < SVN_SIZE) {
+		return -EINVAL;
+	}
+
+	ret = nrf_modem_at_scanf("AT+CGSN=3",
+				 "+CGSN: \"%" STRINGIFY(SVN_SIZE) "[^\"]",
+				 buf);
+
+	if (ret != 1) {
+		LOG_ERR("Could not get SVN, error: %d", ret);
+		return -EIO;
+	}
+	return 0;
+}
+
+int modem_info_get_batt_voltage(int *val)
+{
+	int ret;
+
+	ret = nrf_modem_at_scanf("AT%XVBAT", "%%XVBAT: %d", val);
+
+	if (ret != 1) {
+		LOG_ERR("Could not get battery voltage, error: %d", ret);
+		return -EIO;
+	}
+	return 0;
+}
+
+int modem_info_get_temperature(int *val)
+{
+	int ret;
+
+	ret = nrf_modem_at_scanf("AT%XTEMP?", "%%XTEMP: %d", val);
+
+	if (ret != 1) {
+		LOG_ERR("Could not get temperature, error: %d", ret);
+		return -EIO;
+	}
+	return 0;
+}
+
+int modem_info_get_rsrp(int *val)
+{
+	int ret;
+
+	ret = nrf_modem_at_scanf("AT+CESQ",
+				 "+CESQ: %*d,%*d,%*d,%*d,%*d,%d", val);
+
+	if (ret != 1) {
+		LOG_ERR("at_scanf_int failed");
+		return -EIO;
+	}
+
+	if (*val == 255) {
```

Comment:
Could use some symbol here to represent invalid RSRP value.

Reply:
done